pydra/engine/helpers.py

```diff
@@ -94,7 +95,7 @@ def load_result(checksum, cache_locations):
     return None
 
 
-def save(task_path: Path, result=None, task=None):
+def save(task_path: Path, result=None, task=None, use_locfile=False):
```

Comment:
any particular reason to make this an argument instead of always using it? then you don't need to make the recursive call.

Reply:
i just noticed that in `Task.run` the `SoftFileLock` is already used, but not sure if it's a good enough reason, can probably change